### ISSUE TITLE
Implement ILIKE predicate

### ISSUE THREAD

**Issue by RosterIn** [prestosql/presto#2491]
Comparing case-insensitive strings can be source for confusion.
we must use lower() on both sides of the query like shown:
https://stackoverflow.com/questions/42850329/how-to-configure-presto-searches-to-be-case-insensitive

Some DBs like PostgreSQL extended to support also ILIKE 

> The key word ILIKE can be used instead of LIKE to make the match case-insensitive according to the active locale. This is not in the SQL standard but is a PostgreSQL extension.
> 
> The operator ~~ is equivalent to LIKE, and ~~ * corresponds to ILIKE. There are also !~~ and !~~* operators that represent NOT LIKE and NOT ILIKE, respectively. All of these operators are PostgreSQL-specific.

https://www.postgresql.org/docs/current/functions-matching.html

This is very useful. I hope this can be added to Presto.

**Comment by dain** [comment]
I know it isn't the same, but Presto does support case insensitive regular expression matching with `(?i)` flag. See https://prestosql.io/docs/current/functions/regexp.html for more information.

**Comment by RosterIn** [comment]
so maybe on the backside the ILIKE can be implemented with regexp ?

**Comment by dain** [comment]
The existing `LIKE` is implemented with a regular expression, so if `ILIKE` is added I would expect the same.

**Comment by ebyhr** [comment]
Adding the optional keyword might be one of the candidates. This is Teradata syntax to control the case sensitivity. 
```sql
SELECT empname FROM tbl_emp WHERE empname LIKE '%JO%’;
SELECT empname FROM tbl_emp WHERE empname (CASESPECIFIC) LIKE '%JO%’;
```

**Comment by dain** [comment]
We also could just add a function.

**Comment by martint** [comment]
Yes, a function would be more appropriate than adding dedicated non-standard syntax.

**Comment by KevinKobi** [comment]
I wonder why PostgreSQL choose ILIKE over a function in this case. They must had a reason

**Comment by boristyukin** [comment]
it is also ILIKE in Impala and SnowFlake if it is worth something

**Comment by RosterIn** [comment]
@martint @ebyhr Does the "syntax needs review" label means that only after someone submit PR you will check if this idea is good or not? If there is no intention of accepting this there is no reason to keep this issue alive.

**Comment by ebyhr** [comment]
@RosterIn We use `syntax needs review` label when the syntax isn't yet decided. It doesn't mean we don't accept the PR.

**Comment by zack-prefix** [comment]
This would be incredible!  Here's the state of play on ILIKE (thanks claude):

Engines that support ILIKE natively

PostgreSQL
Amazon Redshift
Snowflake 
Databricks SQL (and PySpark’s .ilike) 
CockroachDB 
DuckDB
ClickHouse
Vertica

Engines that do not have ILIKE

BigQuery 
Trino/Presto 
MySQL / MariaDB 
SQLite
SQL Server 
Oracle

**Comment by bvolpato** [comment]
I've kicked a draft PR with a proposal for this (a function -- hidden, but overall as part of the grammar https://github.com/trinodb/trino/pull/27364).

Is there any desire in following up with ILIKE, given the overall broad support?

**Comment by findepi** [comment]
per @martint 's https://github.com/trinodb/trino/pull/27364#issuecomment-3569169084 we're not currently willing to add `ILIKE` as a new syntax, so the functionality would be implemented as a function.
i synced with @martint and the current plan is to implement the functionality as functions with following signatures

```
ilike(value, patter) -> boolean
ilike(value, patter, scape) -> boolean
```

**Comment by jasonxwlin** [comment]
Does ILIKE syntax perform better than tolower match from performance-wise perspective?
